Clarify the decision making process for version substitution (#1211)
grafana/writers-toolkit: docs/sources/write/links/_index.md

Use a URL.

If you're linking to versioned documentation within your own documentation, use a URL with version substitution syntax instead of the version path element.
For example, in Grafana, use `<GRAFANA_VERSION>` instead of `latest` in the URL `https://grafana.com/docs/grafana/latest/`.

> If you're linking to versioned documentation in another project, use version substitution if you have version compatibility constraints between the projects.

Author: ```suggestion
If you're linking to versioned documentation in another project, use version substitution if you have version compatibility constraints between the projects.
Refer to the target project in the version substitution.
For example, when linking to Tempo documentation from Grafana, use `<TEMPO_VERSION>`.
```

> If you don't have those constraints, use `latest` in the URL.

Reviewer: It's probably worth mentioning here that if you're linking to another project, the version should be _for the project you are linking to_.  I had to fix a couple of broken links to Loki content that used <GRAFANA_VERSION> instead of <LOKI_VERSION> as the substitution.

Author: Applied in e7025922a1e2b38469689c3ba465579a019389a4

Use a URL.

If you're linking to versioned documentation within your own documentation, use a URL with version substitution syntax instead of the version path element.
For example, in Grafana, use `<GRAFANA_VERSION>` instead of `latest` in the URL `https://grafana.com/docs/grafana/latest/`.

If you're linking to versioned documentation in another project, use version substitution if you have version compatibility constraints between the projects.
Refer to the target project in the version substitution.

> For example, when linking to Tempo documentation from Grafana, use `<TEMPO_VERSION>`.

Reviewer 2: Let's see what we can come up with. I like the explanation you have in your comment. I'll see if I can incorporate that into a suggestions.

> If you're linking to versioned documentation in another project, use version substitution if you have version compatibility constraints between the projects.
> Refer to the target project in the version substitution.
> For example, when linking to Tempo documentation from Grafana, use `<TEMPO_VERSION>`.

Reviewer 2: ```suggestion
If you're linking to versioned documentation in another project, use version substitution if you have version compatibility constraints between the projects.
Refer to the target project in the version substitution.

You only need version substitution if you're maintaining a specific matrix of version mappings. 
For example, the Mimir Helm chart for a given release targets supports working only with a specific version of Mimir and Grafana Enterprise Metrics: [grafana/mimir@mimir-distributed-release-5.7/docs/sources/helm-charts/mimir-distributed/_index.md?plain=1#L11-L12](https://github.com/grafana/mimir/blob/mimir-distributed-release-5.7/docs/sources/helm-charts/mimir-distributed/_index.md?plain=1&rgh-link-date=2025-08-05T16%3A16%3A51Z#L11-L12).
In this case, when linking to the Mimir documentation from Grafana, use `<MIMIR_VERSION>`.
```

Use a URL.

If you're linking to versioned documentation within your own documentation, use a URL with version substitution syntax instead of the version path element.
For example, in Grafana, use `<GRAFANA_VERSION>` instead of `latest` in the URL `https://grafana.com/docs/grafana/latest/`.

If you're linking to versioned documentation in another project, use version substitution if you have version compatibility constraints between the projects.
Refer to the target project in the version substitution.

You only need version substitution if you're maintaining a specific matrix of version mappings. 

> For example, the Mimir Helm chart for a given release targets supports working only with a specific version of Mimir and Grafana Enterprise Metrics: [grafana/mimir@mimir-distributed-release-5.7/docs/sources/helm-charts/mimir-distributed/_index.md?plain=1#L11-L12](https://github.com/grafana/mimir/blob/mimir-distributed-release-5.7/docs/sources/helm-charts/mimir-distributed/_index.md?plain=1&rgh-link-date=2025-08-05T16%3A16%3A51Z#L11-L12).

Author: ```suggestion
For example, the Mimir Helm chart for a given release targets supports working only with a specific version of Mimir and Grafana Enterprise Metrics: [`grafana/mimir@mimir-distributed-release-5.7/docs/sources/helm-charts/mimir-distributed/_index.md`](https://github.com/grafana/mimir/blob/mimir-distributed-release-5.7/docs/sources/helm-charts/mimir-distributed/_index.md?plain=1&rgh-link-date=2025-08-05T16%3A16%3A51Z#L11-L12).
```
